apps/central-scan/frontend/src/screens/ballot_eject_screen.tsx

```diff
@@ -222,39 +214,134 @@ export function BallotEjectScreen({ isTestMode }: Props): JSX.Element | null {
           }
         }
       }
-    } else {
-      isUnreadableSheet = true;
     }
   }
 
-  const allowBallotDuplication =
-    !isInvalidTestModeSheet &&
-    !isInvalidBallotHashSheet &&
-    !isUnreadableSheet &&
-    !(isOvervotedSheet && precinctScanDisallowCastingOvervotes);
-
   const backInterpretation = reviewInfo.interpreted.back.interpretation;
   const isBackIntentionallyLeftBlank =
     backInterpretation.type === 'InterpretedHmpbPage' &&
     backInterpretation.markInfo.marks.length === 0;
   const isBlankSheet =
     isFrontBlank && (isBackBlank || isBackIntentionallyLeftBlank);
 
-  const adjudicationTitle = isInvalidTestModeSheet
-    ? isTestMode
-      ? 'Official Ballot'
-      : 'Test Ballot'
-    : isInvalidBallotHashSheet
-    ? 'Wrong Election'
-    : isUnreadableSheet
-    ? 'Unreadable'
-    : isOvervotedSheet
-    ? 'Overvote'
-    : isBlankSheet
-    ? 'Blank Ballot'
-    : isUndervotedSheet
-    ? 'Undervote'
-    : 'Unknown Issue';
+  if (isOvervotedSheet) {
+    for (const contestId of overvoteContestIds) {
+      highlightedContestIds.add(contestId);
+    }
+  } else if (isUndervotedSheet && !isBlankSheet) {
+    for (const contestId of undervoteContestIds) {
+      highlightedContestIds.add(contestId);
+    }
+  }
+
+  const ejectInfo: EjectInformation = (() => {
```

Comment:
This screen and it's confusing logic and driven me crazy for a while so, even though this may not have been the ideal PR to refactor, I had to do a little refactoring.

The only difference in the logic is that there is now no fallthrough case "Unknown Issue", that's just collapsed with "Unreadable." Previously that case didn't make much sense anyway, because it would say "Unknown Issue" and then allow you to tabulate the ballot.